src/System.Runtime.Extensions/tests/System/IO/PathTestsBase.cs

```diff
@@ -204,6 +204,165 @@ public partial class PathTestsBase
 
+        public static TheoryData<string, string, string> GetFullPath_Windows_NonFullyQualified => new TheoryData<string, string, string>
+        {
+            { @"C:\git\corefx", @"C:\git\corefx", @"C:\git\corefx" },
+            { @"C:\git\corefx.\.\.\.\.\.", @"C:\git\corefx", @"C:\git\corefx" },
+            { @"C:\git\corefx" + new string(Path.DirectorySeparatorChar, 3) + ".", @"C:\git\corefx", @"C:\git\corefx" },
+            { @"C:\git\corefx\..\" + Path.GetFileName(@"C:\git\corefx") + @"\.\..\" + Path.GetFileName(@"C:\git\corefx"), @"C:\git\corefx", @"C:\git\corefx" },
+            { @"C:\somedir\..", @"C:\git\corefx", @"C:\" },
+            { @"C:\", @"C:\git\corefx", @"C:\" },
+            { @"..\..\..\..", @"C:\git\corefx", @"C:\" },
+            { @"C:\" + new string(Path.DirectorySeparatorChar, 3), @"C:\git\corefx", @"C:\" },
+        };
+
+        public static TheoryData<string, string, string> GetFullPath_CommonUnRootedWindowsData => new TheoryData<string, string, string>
+        {
+            { "", @"C:\git\corefx", @"C:\git\corefx" },
+            { "..", @"C:\git\corefx", Path.GetDirectoryName(@"C:\git\corefx") },
+
+            // Current drive rooted
+            { @"\tmp\bar", @"C:\git\corefx", @"C:\tmp\bar" },
+            { @"\.\bar", @"C:\git\corefx", @"C:\bar" },
+            { @"\tmp\..", @"C:\git\corefx", @"C:\" }, 
+            { @"\tmp\bar\..", @"C:\git\corefx", @"C:\tmp" },
+            { @"\tmp\bar\..", @"C:\git\corefx", @"C:\tmp" },
+            { @"\", @"C:\git\corefx", @"C:\" },
+
+            // Specific drive rooted
+            { @"C:tmp\foo\..", @"C:\git\corefx", @"C:\git\corefx\tmp" },
+            { @"C:tmp\foo\.", @"C:\git\corefx", @"C:\git\corefx\tmp\foo" },
+            { @"C:tmp\foo\..", @"C:\git\corefx", @"C:\git\corefx\tmp" },
+            { @"C:tmp", @"C:\git\corefx", @"C:\git\corefx\tmp" },
+            { @"C:", @"C:\git\corefx", @"C:\git\corefx" },
+            { @"C", @"C:\git\corefx", @"C:\git\corefx\C" },
+
+            { @"Z:tmp\foo\..", @"C:\git\corefx", @"Z:\tmp" },
+            { @"Z:tmp\foo\.", @"C:\git\corefx", @"Z:\tmp\foo" },
+            { @"Z:tmp\foo\..", @"C:\git\corefx", @"Z:\tmp" },
+            { @"Z:tmp", @"C:\git\corefx", @"Z:\tmp" },
+            { @"Z:", @"C:\git\corefx", @"Z:\" },
+            { @"Z", @"C:\git\corefx", @"C:\git\corefx" + Path.DirectorySeparatorChar + @"Z" }
+        };
+
+
+        public static TheoryData<string, string, string> GetFullPath_Windows_UNC => new TheoryData<string, string, string>
+        {
+            { @"foo", @"", @"foo" },
+            { @"foo", @"server1", @"server1\foo" },
+            { @"\foo", @"server2", @"server2\foo" },
+            { @"foo", @"server3\", @"server3\foo" },
+            { @"foo", @"server3\\", @"server3\foo" },
+            { @"..\foo", @"server4", @"server4\..\foo" },
+            { @".\foo", @"server5\share", @"server5\share\foo" },
+            // { @"..\foo", @"server6\share", @"server6\share\foo" },
```

Comment:
That is the correct behavior when uncommented. It should match what you get when you pass it joined to `GetFullPathName()`. As I mentioned to you, there are a bunch of tests for this in [WInterop](https://github.com/JeremyKuhne/WInterop/blob/755f6172528bd3c498382308af61bf71e1894c83/src/Tests/Tests.Shared/FileManagement/FileManagementBehaviors.cs#L172) that you can reference and/or add to to see behavior.

To be super sure, I quickly added another test `InlineData(@"\\Server\Share\..\foo", @"\\Server\Share\foo"),`.